crates/holochain/src/conductor/handle.rs

```diff
@@ -1653,59 +1622,24 @@ impl<DS: DnaStore + 'static> ConductorHandleImpl<DS> {
         cell_ids
     }
 
-    pub(super) fn p2p_env(&self, space: Arc<KitsuneSpace>) -> DbWrite<DbKindP2pAgentStore> {
-        let mut p2p_env = self.p2p_env.lock();
-        let db_sync_strategy = self.db_sync_strategy;
-        p2p_env
-            .entry(space.clone())
-            .or_insert_with(move || {
-                let root_env_dir = self.root_env_dir.as_ref();
-                DbWrite::open_with_sync_level(
-                    root_env_dir,
-                    DbKindP2pAgentStore(space),
-                    match db_sync_strategy {
-                        DbSyncStrategy::Fast => DbSyncLevel::Off,
-                        DbSyncStrategy::Resilient => DbSyncLevel::Normal,
-                    },
-                )
-                .expect("failed to open p2p_agent_store database")
-            })
-            .clone()
+    pub(super) fn p2p_env(&self, hash: &DnaHash) -> DbWrite<DbKindP2pAgentStore> {
+        self.conductor
+            .spaces
+            .p2p_env(hash)
+            .expect("failed to open p2p_agent_store database")
```

Comment:
These are our own internal APIs, why are we bothering having them return a result if we're just going to expect here?

Reply:
This is test-only code, but we also use `conductor.spaces` in real code